src/solver/conv_hip_implicit_gemm_3d_grouped_fwd_xdlops.cpp

```diff
@@ -0,0 +1,484 @@
+                                                          args.strides,
+                                                          args.dilation,
+                                                          args.lPadding,
+                                                          args.rPadding,
+                                                          {},
+                                                          {},
+                                                          {});
+    return conv_ptrs[i]->IsSupportedArgument(argument_ptr.get());
+}
+
+template <typename DataType>
+bool ConvHipImplicitGemm3DGroupFwdXdlops::CheckCKApplicability(
+    const ProblemDescription& problem) const
+{
+    const auto conv_ptrs = DeviceOpGFwdPtrs<DataType>::GetInstances();
+    assert(!conv_ptrs.empty());
+    const auto args = CKArgs{problem};
+    for(int i = 0; i < conv_ptrs.size(); i++)
+    {
+        auto argument_ptr = conv_ptrs[i]->MakeArgumentPointer(nullptr,
+                                                              nullptr,
+                                                              {},
+                                                              nullptr,
+                                                              args.input,
+                                                              args.in_strides,
+                                                              args.weight,
+                                                              args.wei_strides,
+                                                              {},
+                                                              {},
+                                                              args.output,
+                                                              args.out_strides,
+                                                              args.strides,
+                                                              args.dilation,
+                                                              args.lPadding,
+                                                              args.rPadding,
+                                                              {},
+                                                              {},
+                                                              {});
+        if(conv_ptrs[i]->IsSupportedArgument(argument_ptr.get()))
+            return true;
+    }
+    return false;
+}
+
+namespace {
+
+template <typename DataType>
+void RunCKSolution(const Handle& handle,
+                   const AnyInvokeParams& primitive_parameters,
+                   const CKArgs& args,
+                   const std::string& kernel_id)
+{
+    const auto conv_ptrs = DeviceOpGFwdPtrs<DataType>::GetInstances();
+    int i                = 0;
+    for(; i < conv_ptrs.size(); i++)
+    {
+        if(conv_ptrs[i]->GetTypeString() == kernel_id)
+        {
+            break;
+        }
+    }
+    assert(i != conv_ptrs.size());
+    auto& conv_ptr      = conv_ptrs.at(i);
```

Comment:
I think that computation of `conv_ptr` should be moved to GetSolution or to InvokerFactory at least. See https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2305#discussion_r1290087188 for more info.

Reply:
Fyi: I was also trying to move `conv_ptr` computation out of the invoker lambda and it leads to compilation errors that can't be resolved without major changes. See https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2286#pullrequestreview-1578557778